tests/ink-erc20/src/processor.ts

```diff
@@ -0,0 +1,39 @@
+import {SubstrateBatchProcessor, BatchContext} from '@subsquid/substrate-processor'
+import {lookupArchive} from '@subsquid/archive-registry'
+import {ERC20} from './mapping'
+import {db, Store} from './db'
+import {EntityBuffer} from './entityBuffer'
+import {Block} from './model'
+
+const processor = new SubstrateBatchProcessor()
+    .setDataSource({
+        archive: lookupArchive('shibuya', {type: 'Substrate'}),
+    })
+    .addContractsContractEmitted(ERC20.address, {
+        data: {
+            event: {args: true}
+        },
+    })
+
+processor.run(db, async (ctx: BatchContext<Store, any>) => {
+    for (let {header: block, items} of ctx.blocks) {
+        EntityBuffer.add(
+            new Block({
+                id: block.id,
+                number: block.height,
+                timestamp: new Date(block.timestamp),
+            })
+        )
+        for (let item of items) {
+
+            if (item.kind == 'event' && item.event.name == 'Contracts.ContractEmitted') {
+                if (item.event.args.contract == ERC20.address) {
+                    ERC20.parse(block, item.event)
+                }
+            }
+        }
+    }
+    for (let e of EntityBuffer.flush()) {
+        await ctx.store.insert(e)
```

Comment:
the global object for storing all entities looks like a hack. it causes higher count of db requests in comparison with an approach that groups entities and inserts them as a single request

Reply:
wdym? It's the same approach as before, `e` is array of entities, I can rename it to make it more obvious

Reply:
> the global object for storing all entities looks like a hack
yes, but it's the easiest  the to do it

Reply:
> It's the same approach as before
oh, i just didn't pay enough attention for it earlier. let's stay with it for now and just keep it in mind as a potential place for future optimizations